framework/web/Response.php

```diff
@@ -488,6 +485,9 @@ public function sendContentAsFile($content, $attachmentName, $options = [])
             $this->content = $content;
         }
 
+        $mimeType = isset($options['mimeType']) ? $options['mimeType'] : 'application/octet-stream';
+        $this->setDownloadHeaders($attachmentName, $mimeType, !empty($options['inline']), $end - $begin + 1);
```

Comment:
As a result of `getHttpRange`, `$end < StringHelper::byteLength($content)` and `$begin >=0`
Can you show any case when `StringHelper::byteSubstr($content, $begin, $end - $begin + 1)` returns content with a byteLength no equal to `$end - $begin + 1`?